### PR TITLE
One percent

### PR DESCRIPTION
**Pull request by michaelJwilson**
Necessary changes to master just to get the code up and running.  I was going to try to dust off the OBSCONDITIONS dependence, there's already one small function to start this.  Then hopefully better handle LYA-QSO -> QSO etc after redshift, which I think is currently ignored.

### CODE REVIEW

**Comment by michaelJwilson** [comment]
Ok, now non-minimal changes to handle the obsconditions dependence (basically, handle the new exposures and tile formats - like of NIGHT etc.) and repeated TILEIDs due to cosmic splits, non-standard footprint (for one percent) and other arguments to the new fiberassign.

It's possible I broke something and some changes may not have been strictly necessary - e.g. I may have broke and then fixed-up the night format parsing.  But it runs and is probably no more buggy than before.  I'm lacking a good sanity checking example to really make sure.

I'm largely just keeping you in the loop for feedback and oversight.  Feel free to ignore!

**Comment by forero** [comment]
What are you using as inputs?

**Comment by michaelJwilson** [comment]
Everything is here:
/global/cscratch1/sd/mjwilson/svdc2019c2/svdc2019c2, in particular there's a README there with the quicksurvey (and other calls).  In particular, the tiling footprint for the one-percent CENTERIDs and Eddie's new tiling strategy (non-optimized), corresponding exposures from a surveys run with this file and similarly a targets.fits from a select_mock_target run for this.  It's possible not all mock_targets heal pixels got processed, I still need to check this.  NERSC is down from 8 am, so there'll be no progress until next week.  Have a good weekend!

File: bin/wrap-fastframe [hunk 1]
```diff
@@ -67,19 +68,40 @@ if rank == 0:
...
+                if args.outdir:
+                  ##  If outdir provided, log files inherit /night/exposure structure of simspec files but in outdir.                                                                                          
+                  ##  Else placed in original dir. with simspecs.                                                                                                                                              
+
+                  logfile = logfile.replace(desisim.io.simdir(), args.outdir)
+                
+                try:
+                  os.makedirs(os.path.dirname(logfile), exist_ok=args.clobber)
+                except:
+                  print('Pass --clobber to replace previously created logfile for {}.'.format(expid))
```
Comment (sbailey) [inline bin/wrap-fastframe]: Please use 4-space indentation instead of 2-space indentation lines 82-91 (and elsewhere)

File: bin/wrap-fastframe [hunk 2]
```diff
@@ -130,7 +152,11 @@ if rank < len(simspecfiles):
                 ##  If outdir provided, log files inherit /night/exposure structure of simspec files but in outdir. 
                 ##  Else placed in original dir. with simspecs.
                 logfile = logfile.replace(desisim.io.simdir(), args.outdir)
-                os.makedirs(os.path.dirname(logfile), exist_ok=args.clobber)
+
+            try:
+              os.makedirs(os.path.dirname(logfile), exist_ok=args.clobber)
+            except:
+              print('Pass --clobber to replace previously created logfile: {}.'.format(logfile))
```
Comment (sbailey) [inline bin/wrap-fastframe]: 2-space indentation -> 4-space indentation

File: py/desisim/quickcat.py [hunk 3]
```diff
@@ -606,26 +669,28 @@ def quickcat(tilefiles, targets, truth, zcat=None, obsconditions=None, perfect=F
...
-    #- Trim truth down to just ones that have already been observed
+    #- prevent future surprises if code expects them to be row aligned.
+    tileids       = np.array(tileids)
+
+    if (obsconditions is not None):
+      ##  Handle repeated TILEIDs for Cosmic splits.
+      order       = [list(np.where(obsconditions['TILEID'].quantity == x)[0]) for x in tileids]
+      order       = [item for sublist in order for item in sublist]    
+
+      obsconditions = obsconditions[order]
```
Comment (sbailey) [inline py/desisim/quickcat.py]: 2-space indentation -> 4-space indentation

File: py/desisim/quicksurvey.py [hunk 4]
```diff
@@ -75,41 +81,50 @@ def __init__(self, output_path, targets_path, fiberassign, exposures, fiberassig
...
         self.n_epochs = len(dates) - 1
 
+        ##  Convert MJD present in exposures to 'NIGHT' in yyyymmdd format.                                                                                                                                     
+        iso     = Time(self.exposures['MJD'], format='mjd', scale='utc').iso
+        dateobs = np.array([x.split(' ')[0].replace('-', '') for x in iso]).astype(np.int)
+        
+        #- add pre- and post- dates for date range bookkeeping                                                                                                                                                
+        if dates[0] < min(dateobs):
+          dates.insert(19000131, dateobs[0])
+          dates.append(30000131)
```
Comment (sbailey) [inline py/desisim/quicksurvey.py]: 2-space indentation -> 4-space indentation

**Comment by moustakas** [comment]
Is this PR still active / valid? @michaelJwilson

**Comment by michaelJwilson** [comment]
Yes, in the sense that I have a new version from the last few weeks that usurps this.  Amongst other things, it handles:

--  Non-standard footprints. 
--  Assigning epochs according to actual fiberassign date extracted from surveysim, rather than guessing on exposure completion.   Propagation of RUNDATE to fiberassign calls.  
--  Generation of fibermap files. 
-- etc. 

An example run lives here:
/global/homes/m/mjwilson/desi/survey-validation/svdc-spring2020d-onepercent/

I'm coordinating with  @ashleyjross  to build something for the clustering groups that has a useable LSS catalog built upon the relevant pipeline files.  This PR should wait until we've converged on that.  I can close it in the meantime if that's useful?

**Comment by sbailey** [comment]
@michaelJwilson if I understand correctly, you will soon submit a new PR that supersedes this one.  If that is correct, please go ahead and close this PR without merging.  Thanks.

**Pull request closed without merging**